docs/AsyncMoya.md

```diff
@@ -0,0 +1,68 @@
+# Async/await
+
+Moya provides an optional `Async/await` implementation of
+`MoyaProvider` that does a few interesting things. Instead of
+calling the `request()` method and providing a callback closure
+to be executed when the request completes, we use `await`.
+
+```swift
+let provider = MoyaProvider<GitHub>()
+```
+
+After that simple setup, you're off to the races:
+
+```swift
+AsyncTask {
+    let result = await provider.asyncRequest(.zen) //return type `Result<Response, MoyaError>`
+    switch result {
+    case let .success(response):
+        // do something with the data
+    case let .failure(error):
+        // handle the error
+    default:
+        break
+    }
+}
+```
+
+`AsyncTask` is `typealias` for Apple mechanism `Task` for resolve conflict with `Moya.Task`.
```

Comment:
I think it may be worth thinking about considering to rename `Moya.Task`, as this is (?) referring to another construct than `Task` in an async context, and may prove confusing going forward. What do you think?

Reply:
yes, i think Moya.Task should be renamed, but this will "breaking changes" for all and i haven't idea how to do it better now :(

Reply:
We could at least think of moving away/renaming `Moya.Task`. You could create an issue for that if you wish. :)

Reply:
At least deprecating `Moya.Task` as well may be a good thing to do here, so that in a future (major) release, we can move to the desired naming (and use `Swift.Task` directly).

Reply:
good idea 👍🏻 
i can mark `Moya.Task` like deprecate in future version, so that users are ready)

Reply:
done ✅